lib/demux/transmitter.rb

```diff
@@ -48,9 +53,30 @@ def build_request
 
     def send_request
       @response =
-        Net::HTTP.start(@uri.hostname, @uri.port, use_ssl: true) do |http|
+        Net::HTTP.start(@uri.hostname, @uri.port, **request_options) do |http|
           http.request(@request)
         end
+
+      self
+    rescue Net::ReadTimeout, Net::OpenTimeout, Net::WriteTimeout
```

Comment:
Net::HTTP breaks timeouts into three possibilities... Read, Open, and Write. For our initial purposes, we are just treating them all the same. This was determined by searching the Net::HTTP docs for "timeout" https://ruby-doc.org/stdlib-2.7.1/libdoc/net/http/rdoc/Net/HTTP.html

For some inspiration, I looked at how httparty treats timeouts: https://github.com/jnunemaker/httparty/blob/b9a54d8f73a9a94863bf83a1ba559b557c68b4c8/lib/httparty/connection_adapter.rb#L117

It gives you the option to set all at once using "timeout" or individually. I don't see a need for individually values here currently.

Reply:
> I don't see a need for individually values here currently.

Agreed. Your solution seems like a great starting point for us to catch and identify timeouts